command/agent/txn_endpoint.go

```diff
@@ -0,0 +1,227 @@
+package agent
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/consul/structs"
+)
+
+const (
+	// maxTxnOps is used to set an upper limit on the number of operations
+	// inside a transaction. If there are more operations than this, then the
+	// client is likely abusing transactions.
+	maxTxnOps = 500
```

Comment:
I would consider limiting the number of operations and the size of the txn as well. This is to avoid having 64 PUTs of 512KB or something.

Reply:
Do you guys still think 64 is still valid? Our system has about 100 env (between configs and feature flags). I think we can either remove the limit all together or at least have it configurable. You can see it has been requested [here](https://github.com/hashicorp/consul/issues/2921) and [here](https://github.com/hashicorp/consul/issues/3917)

Thanks

Reply:
@mostafazh It should be made configurable or the system enhanced to handle arbitrarily large transactions. As it stands, a transaction that is too large would get rejected by the underlying Raft layer, and below that threshold could cause unexpected performance issues and head of line blocking. For most users, those tradeoffs would not be clear (or expected) in the configuration, which is why it is not currently configurable.